benchmarks/smallmatvec/manual_versions/libxsmm/kdriver.f90

```diff
@@ -0,0 +1,186 @@
+program kdriver
+  use constants_mod, only : i_def, r_def, str_max_filename
+  use matrix_vector_kernel_mod
+  use omp_lib
+  implicit none
+  ! the arguments and array sizes needed to read/write the arrays.
+  integer(kind=i_def) :: ndf_any_space_1_theta_adv_term, undf_any_space_1_theta_adv_term, ndf_any_space_2_x, undf_any_space_2_x
+  integer(kind=i_def) :: ncell, nlayers, ncell_3d, ncolour
+  integer(kind=i_def), allocatable :: ncp_colour(:), cmap(:,:)
+  integer(kind=i_def), allocatable,target :: map_any_space_1_theta_adv_term(:,:)
+  integer(kind=i_def), allocatable,target :: map_any_space_2_x(:,:)
+  real(kind=r_def), allocatable :: theta_adv_term_data(:), x_data(:), ans_data(:)
+  real(kind=r_def), allocatable :: ptheta_2_local_stencil(:,:,:)
+  real(kind=r_def), allocatable :: nlayers_first(:,:,:,:)
+  integer(kind=i_def) :: df, df2, lp, nthreads, memstart,memend
+  real(kind=r_def) :: diff, err
+  real(kind=r_def) :: start, end
+
+  integer(kind=i_def), pointer :: map1(:) =>null(), map2(:)=>null()
+  integer(kind=i_def) :: colour, cell, idx, tile, r, row
+  integer(kind=i_def) :: c1_index, c2_index, c3_index, c4_index
+
+  integer(kind=i_def) :: Nreps, nsize, el, i1, i2, i
+  character(len=32) :: arg, arg2, initialization, method, traverse, colouringkind 
+  logical :: new_data_layout
+
+  real(kind=r_def), dimension(144) :: SFC12, SFC12i
+
+  integer(kind = OMP_lock_kind), allocatable :: lock(:,:)
+
+  traverse = "colouring" ! Default traversing
+  ! cmdline argument parsing
+  do i = 1, command_argument_count()
+    call get_command_argument(i, arg)
+
+    select case (arg)
+    case ('-g', '--generate')
+        initialization="generate"
```

Comment:
It looks as though this code will always 'generate' so is this cmd-line parsing necessary?

Reply:
Simplified.